apps/web-giddh/src/app/inventory/components/stock-report-component/inventory.stockreport.component.ts

```diff
@@ -1,328 +1,330 @@
-		this.stockReportRequest = new StockReportRequest();
-		this.universalDate$ = this.store.select(p => p.session.applicationDate).pipe(takeUntil(this.destroyed$));
-		this.entityAndInventoryTypeForm = this.fb.group({
-			selectedEntity: ['allEntity'],
+    /**
+     * TypeScript public modifiers
+     */
+    constructor(private store: Store<AppState>, private route: ActivatedRoute, private sideBarAction: SidebarAction,
+                private stockReportActions: StockReportActions, private router: Router,
+                private _toasty: ToasterService,
+                private inventoryService: InventoryService, private fb: FormBuilder, private inventoryAction: InventoryAction,
+                private settingsBranchActions: SettingsBranchActions,
+                private invViewService: InvViewService,
+                private cdr: ChangeDetectorRef
+    ) {
+        this.stockReport$ = this.store.select(p => p.inventory.stockReport).pipe(takeUntil(this.destroyed$), publishReplay(1), refCount());
+        this.stockReportRequest = new StockReportRequest();
+        this.universalDate$ = this.store.select(p => p.session.applicationDate).pipe(takeUntil(this.destroyed$));
+        this.entityAndInventoryTypeForm = this.fb.group({
+            selectedEntity: ['allEntity'],
             selectedTransactionType: ['all']
-		});
-	}
-
-	public findStockNameFromId(grps: IGroupsWithStocksHierarchyMinItem[], stockUniqueName: string): string {
-		if (grps && grps.length > 0) {
-			for (let key of grps) {
-				if (key.stocks && key.stocks.length > 0) {
-
-					let index = key.stocks.findIndex(p => p.uniqueName === stockUniqueName);
-					if (index === -1) {
-						let result = this.findStockNameFromId(key.childStockGroups, stockUniqueName);
-						if (result !== '') {
-							return result;
-						} else {
-							continue;
-						}
-					} else {
-						this.activeStockName = key.stocks[index].name;
-						return key.stocks[index].name;
-					}
-				} else {
-					let result = this.findStockNameFromId(key.childStockGroups, stockUniqueName);
-					if (result !== '') {
-						return result;
-					} else {
-						continue;
-					}
-				}
-			}
-			return '';
-		}
-		return '';
-	}
-
-	public ngOnInit() {
-		if (this.route.firstChild) {
-			this.route.firstChild.params.pipe(take(1)).subscribe(s => {
-				if (s) {
-					this.groupUniqueName = s.groupUniqueName;
-					this.stockUniqueName = s.stockUniqueName;
-					this.initReport();
-				}
-			});
-        }
-
-		// get view from sidebar while clicking on group/stock
-		this.invViewService.getActiveView().subscribe(v => {
-			this.initVoucherType();
-			this.groupUniqueName = v.groupUniqueName;
-			this.stockUniqueName = v.stockUniqueName;
-			this.selectedEntity = 'allEntity';
+        });
+        this.UpdateStockSuccess$ = this.store.select(s => s.inventory.UpdateStockSuccess).pipe(takeUntil(this.destroyed$));
```

Comment:
Consider using `this.store.pipe(select(...));` as `this.store.select` is deprecated